### PR TITLE
Split infinispan testing into separate modules for the quarkus and camel managed clients

### PR DESCRIPTION
**Pull request by jamesnetherton**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: extensions/infinispan/deployment/src/main/java/org/apache/camel/quarkus/component/infinispan/deployment/InfinispanProcessor.java [hunk 1]
```diff
@@ -27,4 +30,16 @@ class InfinispanProcessor {
     FeatureBuildItem feature() {
         return new FeatureBuildItem(FEATURE);
     }
+
+    @BuildStep
+    NativeImageResourceBuildItem nativeImageResources() {
+        // Only required when Camel instantiates and manges its own internal CacheContainer
```
Comment (zbendhiba) [inline extensions/infinispan/deployment/src/main/java/org/apache/camel/quarkus/component/infinispan/deployment/InfinispanProcessor.java:36]: I guess there's a typo : manges ==> manages ?

File: extensions/infinispan/deployment/src/main/java/org/apache/camel/quarkus/component/infinispan/deployment/InfinispanProcessor.java [hunk 2]
```diff
@@ -27,4 +30,16 @@ class InfinispanProcessor {
...
+        return new NativeImageResourceBuildItem("org/infinispan/protostream/message-wrapping.proto");
+    }
+
+    @BuildStep
+    ReflectiveClassBuildItem reflectiveClasses() {
+        // Only required when Camel instantiates and manges its own internal CacheContainer
```
Comment (zbendhiba) [inline extensions/infinispan/deployment/src/main/java/org/apache/camel/quarkus/component/infinispan/deployment/InfinispanProcessor.java:42]: same comment here

File: extensions/infinispan/deployment/src/main/java/org/apache/camel/quarkus/component/infinispan/deployment/InfinispanProcessor.java (hunk 1 above)
Comment (jamesnetherton) [reply extensions/infinispan/deployment/src/main/java/org/apache/camel/quarkus/component/infinispan/deployment/InfinispanProcessor.java]: Yes, well spotted :eyes:

File: integration-tests/infinispan-quarkus-client/pom.xml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.rest-assured</groupId>
+            <artifactId>rest-assured</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-integration-test-infinispan-common</artifactId>
```
Comment (zbendhiba) [inline integration-tests/infinispan-quarkus-client/pom.xml:60]: This dependency is pulled twice

File: integration-tests/infinispan/pom.xml [hunk 4]
```diff
@@ -67,6 +56,12 @@
             <artifactId>rest-assured</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-integration-test-infinispan-common</artifactId>
```
Comment (zbendhiba) [inline integration-tests/infinispan/pom.xml:61]: the dependency is pulled twice here too

**Review by zbendhiba — APPROVED** [review]

File: integration-tests/infinispan-quarkus-client/pom.xml (hunk 3 above)
Comment (jamesnetherton) [reply integration-tests/infinispan-quarkus-client/pom.xml:60]: It's intended...

One is the `test-jar` containing 'common' tests. The other is the plain `jar` containing 'common' Camel routes and JAX-RS services.

Comment (zbendhiba) [reply integration-tests/infinispan-quarkus-client/pom.xml:60]: Oh okay

**Review by aldettinger — APPROVED** [review]

**Pull request merged**